tests/ssh/test_ssh_limit.py

```diff
@@ -0,0 +1,150 @@
+    modify_template(admin_session, PAM_LIMITS_TEMPLATE_PATH, additional_content, hwsku, type)
+
+    # Backup and change /usr/share/sonic/templates/limits.conf.j2
+    additional_content = "{0}  hard  maxlogins  1".format(user)
+    modify_template(admin_session, LIMITS_CONF_TEMPLATE_PATH, additional_content, hwsku, type)
+
+def restore_templates(duthost):
+    duthost.shell(TEMPLATE_RESTORE_COMMAND.format(PAM_LIMITS_TEMPLATE_PATH))
+    duthost.shell(TEMPLATE_RESTORE_COMMAND.format(LIMITS_CONF_TEMPLATE_PATH))
+
+def restart_hostcfgd(duthost):
+    duthost.shell(HOSTSERVICE_RELOADING_COMMAND)
+    time.sleep(HOSTSERVICE_RELOADING_TIME)
+
+def limit_template_exist(duthost):
+    return duthost.stat(path=LIMITS_CONF_TEMPLATE_PATH).get('stat', {}).get('exists', False)
+
+@pytest.fixture(scope="module")
+def setup_limit(duthosts, rand_one_dut_hostname, tacacs_creds, creds):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # if template file not exist on duthost, ignore this UT
+    # However still need yield, if not yield, UT will failed with StopIteration error.
+    template_file_exist = limit_template_exist(duthost)
+    if template_file_exist:
+        setup_local_user(duthost, tacacs_creds)
+
+        # Modify templates and restart hostcfgd to render config files
+        modify_templates(duthost, tacacs_creds, creds)
+        restart_hostcfgd(duthost)
+
+        # for debug, print rendered config file
+        config_file_content = duthost.shell('sudo cat /etc/security/limits.conf', module_ignore_errors=False)['stdout_lines']
+        logging.debug("Updated config file: {0}".format(config_file_content))
+
+    yield
+
+    if template_file_exist:
+        # Restore SSH session limit
+        restore_templates(duthost)
+        restart_hostcfgd(duthost)
+
+def get_login_result(ssh_session):
+    login_channel = ssh_session.invoke_shell()
+    login_message = ""
+    start_time = time.time()
+    while (time.time() - start_time) <= LOGIN_MESSAGE_TIMEOUT:
+        if login_channel.recv_ready():
+            data = login_channel.recv(LOGIN_MESSAGE_BUFFER_SIZE)
+            if len(data) == 0:
+                # when receive zero length data, channel closed
+                break
+            login_message += data
+
+        time.sleep(1)
+
+    return login_message
+
+def test_ssh_limits(duthosts, rand_one_dut_hostname, tacacs_creds, setup_limit):
+    """
+        This test case will test following 2 scenarios:
+            1. Following 2 templates can be render by hostcfgd correctly:
+                    /usr/share/sonic/templates/pam_limits.j2
+                    /usr/share/sonic/templates/limits.conf.j2
+            2. SSH login session limit works correctly.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # if template file not exist on duthost, ignore this UT
+    if not limit_template_exist(duthost):
+        logging.debug("Template file {0} not exist, ignore test case.".format(LIMITS_CONF_TEMPLATE_PATH))
+        return
```

Comment:
When template is not there, the test should be skipped. return will make test passed.

Reply:
you can use the pytest_requires  helper.

Reply:
Fixed with pytest.skip.
I checked currently sonic-mgmt not install and use pytest-requires package, follow other test case to minimize code/config change.

Reply:
Fixed again, change to use pytest_require from tests.common.helpers.assertions.